src/Middleware/HttpLogging/test/HttpLoggingMiddlewareTests.cs

```diff
@@ -838,6 +839,125 @@ public async Task UnrecognizedMediaType()
         Assert.Contains(TestSink.Writes, w => w.Message.Contains("Unrecognized Content-Type for body."));
     }
 
+    [Fact]
+    public async Task UpgradeToWebSocketLogsResponsePropertiesAndHeadersWhenResponseIsFlushed()
```

Comment:
Add a negative test for when the WebSocket is not upgraded, the logs should be written normally by the middleware instead of the new upgrade feature.